apps/api/src/app/topics/use-cases/add-subscribers/add-subscribers.use-case.ts

```diff
@@ -1,20 +1,64 @@
 import { TopicSubscribersEntity, TopicSubscribersRepository } from '@novu/dal';
+import { SubscriberDto } from '@novu/shared';
 import { ConflictException, Injectable } from '@nestjs/common';
+import { ExternalSubscriberId } from '../../types';
 
 import { AddSubscribersCommand } from './add-subscribers.command';
 
+import { SearchByExternalSubscriberIds, SearchByExternalSubscriberIdsCommand } from '../../../subscribers/usecases';
+
 @Injectable()
 export class AddSubscribersUseCase {
-  constructor(private topicSubscribersRepository: TopicSubscribersRepository) {}
+  constructor(
+    private searchByExternalSubscriberIds: SearchByExternalSubscriberIds,
+    private topicSubscribersRepository: TopicSubscribersRepository
+  ) {}
 
   async execute(command: AddSubscribersCommand) {
-    const entity = this.mapToEntity(command);
+    const filteredSubscribersCommand = await this.filterExistingSubscribers(command);
 
+    const entity = this.mapToEntity(filteredSubscribersCommand);
```

Comment:
Yet to decide how we will inform in the API which subscribers we could save and which ones didn't. To be done in a different PR.